### PR TITLE
Use HotRod API instead of Spring's @Cacheable

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #5

### CODE REVIEW

**Comment by geoand** [comment]
@Ladicek Ready!

Not the prettiest implementation, but it should work :)
I basically take advantage of Spring only for setting up the cache, the actual interaction with it is done via the native API

File: greeting-service/src/main/java/io/openshift/booster/service/NameCacheUtil.java [hunk 1]
```diff
@@ -29,22 +32,30 @@
...
     public boolean isEmpty() {
         return infispanCache.isEmpty();
     }
+
+    public String get() {
+        return infispanCache.get(KEY);
+    }
+
+    public void put(String value) {
+        infispanCache.put(KEY, value, ttlSeconds, TimeUnit.MILLISECONDS);
```
Comment (Ladicek) [inline greeting-service/src/main/java/io/openshift/booster/service/NameCacheUtil.java]: If `ttlSeconds` is in seconds, then `TimeUnit.MILLISECONDS` is wrong :-)

Comment (Ladicek) [reply greeting-service/src/main/java/io/openshift/booster/service/NameCacheUtil.java]: And I see below that `cache.ttl` is actually in millis -- so it's the variable name `ttlSeconds` what is wrong.

**Comment by Ladicek** [comment]
Could you also please do this? https://github.com/wildfly-swarm-openshiftio-boosters/wfswarm-cache/commit/71e764ad578405b647274f46ea4413a41406b567 Thanks

File: greeting-service/src/main/java/io/openshift/booster/service/NameCacheUtil.java (hunk 1 above)
Comment (geoand) [reply greeting-service/src/main/java/io/openshift/booster/service/NameCacheUtil.java]: Yup, I changed it after fixing the tests, but obviously didn't fix it everywhere :)

File: greeting-service/src/main/java/io/openshift/booster/service/NameCacheUtil.java [hunk 2]
```diff
@@ -29,22 +32,30 @@
 @Component
 public class NameCacheUtil {
 
-    private final Cache springCache;
+    private static final String KEY = "key";
+
+    private final int ttlMillis;
     private final BasicCache<String, String> infispanCache;
```
Comment (Ladicek) [inline greeting-service/src/main/java/io/openshift/booster/service/NameCacheUtil.java]: I just noticed -- this variable is called "infispan", should be "infinispan" :-)

Comment (geoand) [reply greeting-service/src/main/java/io/openshift/booster/service/NameCacheUtil.java]: Haha :P Damn typos :). I'll fix it

**Comment by Ladicek** [comment]
Looks good to me, but the tests fail with things like `Caused by: org.infinispan.commons.CacheConfigurationException: ISPN000432: Missing configuration for default cache 'default' declared on container`.

File: greeting-service/src/main/resources/infinispan-conf.xml [hunk 3]
```diff
@@ -4,9 +4,6 @@
         xmlns="urn:infinispan:config:9.1">
 
     <cache-container name="container" default-cache="default">
-        <local-cache name="default">
-            <expiration lifespan="5000"/>
-        </local-cache>
```
Comment (Ladicek) [inline greeting-service/src/main/resources/infinispan-conf.xml]: You probably should have removed just the `expiration` line, leaving the `local-cache` intact.

Comment (geoand) [reply greeting-service/src/main/resources/infinispan-conf.xml]: You're right, all do that

Comment (geoand) [reply greeting-service/src/main/resources/infinispan-conf.xml]: Done

**Comment by geoand** [comment]
Thank you for your help @Ladicek!

**Comment by Ladicek** [comment]
Thanks for such a quick turn-around!

**Comment by Ladicek** [comment]
And I really like how rhoar-bot helps, when it's stable :-)

**Comment by geoand** [comment]
rhoar-bot is amazing! The fast feedback is invaluable! Great job!!!!

**Pull request merged**